scripts/create_test_subset.py

```diff
@@ -104,19 +103,17 @@ def main(
     pid_sid = papi.get_external_participant_id_to_internal_sample_id(project)
     sample_id_by_participant_id = dict(pid_sid)
 
-    ped_lines = export_ped_file(project, replace_with_participant_external_ids=True)
     if families_n is not None:
-        ped = Ped(ped_lines)
-        families = list(ped.families.values())
-        logger.info(f'Found {len(families)} families, by size:')
-        _print_fam_stats(families)
-        families = random.sample(families, families_n)
-        logger.info(f'After subsetting to {len(families)} families:')
-        _print_fam_stats(families)
-        sample_ids = []
-        for fam in families:
-            for s in fam.samples:
-                sample_ids.append(sample_id_by_participant_id[s.sample_id])
+        fams = fapi.get_families(project)
+        all_families = [family['id'] for family in fams]
+        families = random.sample(all_families, families_n)
+        pedigree = fapi.get_pedigree(project=project, internal_family_ids=families)
+        p_ids = [ped['individual_id'] for ped in pedigree]
+        sample_ids = [
+            sample
+            for (participant, sample) in sample_id_by_participant_id.items()
+            if participant in p_ids
+        ]
         samples = [s for s in all_samples if s['id'] in sample_ids]
```

Comment:
calling `in` on a non-hashed type is an `O(n) operation, and in a loop like this is `O(n**2)`, you might consider making a set version of sample_ids to do a contains check.

Reply:
I think I've followed your suggestion in the latest push but let me know if I misinterpreted